Scripts/public-api-diff/Sources/Helpers/ProjectHelper.swift

```diff
@@ -0,0 +1,169 @@
+            let oldWorkingDirectoryPath = try setupProject(from: oldSource)
+            
+            try body(oldWorkingDirectoryPath, newWorkingDirectoryPath)
+        } catch {
+            cleanup()
+        }
+    }
+}
+
+private extension ProjectHelper {
+
+    func retrieveRemoteProject(branchOrTag: String, repository: String) -> String {
+        
+        let currentDirectory = fileHandler.currentDirectoryPath
+        let targetDirectoryPath = currentDirectory.appending("\(UUID().uuidString)")
+        
+        try? fileHandler.removeItem(atPath: targetDirectoryPath)
+        
+        let git = Git(shell: shell)
+        git.clone(repository, at: branchOrTag, targetDirectoryPath: targetDirectoryPath)
+        return targetDirectoryPath
+    }
+    
+    func setupProject(from source: ProjectSource) throws -> String {
+        
+        let sourceDirectoryPath: String
+        
+        switch source {
+        case let .local(path):
+            sourceDirectoryPath = path
+            
+        case let .remote(branchOrTag, repository):
+            sourceDirectoryPath = retrieveRemoteProject(branchOrTag: branchOrTag, repository: repository)
+        }
+
+        let sourceWorkingDirectoryPath = workingDirectoryPath.appending("/\(UUID().uuidString)")
+        
+        try Self.setupIndividualWorkingDirectory(
+            at: sourceWorkingDirectoryPath,
+            sourceDirectoryPath: sourceDirectoryPath,
+            fileHandler: fileHandler,
+            shell: shell
+        )
+        
+        let packagePath = PackageFileHelper.packagePath(for: sourceWorkingDirectoryPath)
+        
+        try Self.buildProject(
+            projectDirectoryPath: sourceWorkingDirectoryPath,
+            packageFileHelper: PackageFileHelper(packagePath: packagePath, fileHandler: fileHandler),
+            xcodeTools: XcodeTools(shell: shell)
+        )
+        
+        switch source {
+        case .local:
+            break
+        case .remote:
+            // Clean up the cloned repo
+            try? fileHandler.removeItem(atPath: sourceDirectoryPath)
+        }
+        
+        return sourceWorkingDirectoryPath
+    }
+    
+    static func setupIndividualWorkingDirectory(
+        at destinationDirectoryPath: String,
+        sourceDirectoryPath: String,
+        fileHandler: FileHandling,
+        shell: ShellHandling
+    ) throws {
+        
+        try? fileHandler.removeItem(atPath: destinationDirectoryPath)
+        try fileHandler.createDirectory(atPath: destinationDirectoryPath)
+        
+        let fileNameIgnoreList: Set<String> = [
+            ".build",
+            ".git",
+            ".github",
+            ".gitmodules",
+            ".codebeatignore",
+            ".swiftformat",
+            ".swiftpm",
+            ".DS_Store",
+            "Tests",
+            "Cartfile"
+        ]
+        
+        let fileExtensionIgnoreList: Set<String> = [
```

Comment:
Couldn't these two sets be combined?

Reply:
Yeah, not too happy with these lists either - would have to replace it with a regex - I added a TODO to tackle in another PR